src/families/cardano/preload.ts

```diff
@@ -0,0 +1,49 @@
+import { Observable, Subject } from "rxjs";
+import { log } from "@ledgerhq/logs";
+
+import type { CardanoPreloadData } from "./types";
+import { getPreloadedData } from "./api";
+
+const PRELOAD_MAX_AGE = 24 * 60 * 60 * 1000; // 1 day
+
+let currentPreloadedData: CardanoPreloadData;
+
+function fromHydratePreloadData(data: any): CardanoPreloadData {
+  return data;
+}
+
+const updates = new Subject<CardanoPreloadData>();
+
+export function getCurrentCardanoPreloadData(): CardanoPreloadData {
+  return currentPreloadedData;
+}
+
+export function setCardanoPreloadData(data: CardanoPreloadData): void {
+  if (data === currentPreloadedData) return; // TODO: check logic
+  currentPreloadedData = data;
+  updates.next(data);
+}
+
+export function getCardanoPreloadDataUpdates(): Observable<CardanoPreloadData> {
```

Comment:
In other implementations (i.e Polkadot) this is only used to provide a React hook for the UI, see here: https://github.com/LedgerHQ/ledger-live-common/blob/develop/src/families/polkadot/react.ts#L16
You'll probably want such a hook to use the preload data.